src/placeos-models/user.cr

```diff
@@ -161,16 +165,19 @@ module PlaceOS::Model
       User.find_all([auth_id], index: :authority_id)
     end
 
-    secondary_index :email
+    secondary_index :email_digest
 
     def self.find_by_email(authority_id : String, email : String)
-      User.where(email: email, authority_id: authority_id).first?
+      find_by_emails(authority_id, [email]).first?
     end
 
     def self.find_by_emails(authority_id : String, emails : Array(String))
-      User.raw_query do |r|
-        r.table(User.table_name)
-          .get_all(emails, index: :email)
+      return [] of self if emails.empty?
+
+      digests = emails.map &->weak_digest(String)
+      User.collection_query do |table|
+        table
+          .get_all(digests, index: :email_digest)
```

Comment:
I think this [may be deprecated](https://github.com/spider-gazelle/rethinkdb-orm/blob/f0eaf97eb5ceca7fa6ff63f2b9c2d8e60cf01f66/src/rethinkdb-orm/queries.cr#L85-L87).

Not overly familiar with internal by could a `.where` be used here instead (assuming it does the index lookup)?

Reply:
Nope. That's the query dsl.